src/js/tabs/cli.js

```diff
@@ -404,4 +433,305 @@ TABS.cli.cleanup = function (callback) {
         CONFIGURATOR.cliActive = false;
         CONFIGURATOR.cliValid = false;
     });
+
+    this.autoComplete.cleanup();
+};
+
+
+
+
+/**
+ * Encapsulates the AutoComplete logic
+ *
+ * Uses: https://github.com/yuku/jquery-textcomplete
+ * Check out the docs at https://github.com/yuku/jquery-textcomplete/tree/v1/doc
+ *
+ * @param {jquery} $textarea
+ */
+function AutoComplete($textarea) {
+    const self = this;
+
+    self.$textarea = $textarea;
+    self.forceOpen = false,
+    self.cache = {
+        commands: {},
+        resources: [],
+        resourcesCount: {},
+        settings: [],
+        settingsAcceptedValues: {}
+    };
+    self.builder = {
+        state: 'init',
+        stateSequence: [{state: 'parse-commands', command: 'help'},
+                        {state: 'parse-resources', command: 'dump'},
+                        {state: 'parse-get', command: 'get'}],
+        currentSetting: null
+    };
+
+    self._initTextcomplete();
+};
+
+AutoComplete.prototype.isEnabled = function() {
+    return this.builder.state != 'fail';
+};
+
+AutoComplete.prototype.isBuilding = function() {
+    return this.builder.state != 'fail' && this.builder.state != 'done';
+};
+
+AutoComplete.prototype.isOpen = function() {
+    return $('.textcomplete-dropdown').is(':visible');
+};
+
+/**
+ *
+ * @param {boolean} force - Forces AutoComplete to be shown even if the matching strategy has less that minChars input
+ */
+AutoComplete.prototype.open = function(force) {
+    const self = this;
+    setTimeout(function() {
+        self.forceOpen = !!force;
+        self.$textarea.textcomplete('trigger');
+        self.forceOpen = false;
+    }, 0);
+};
+
+AutoComplete.prototype.buildCache = function(line) {
+    const sentinel = '# 131be3f41207bf8b05a253f89d551d4c';
+    const timer = 'autocomplete_building_watchdog';
+    const cache = this.cache;
+    const builder = this.builder;
+    const isSentinel = line && line.indexOf(sentinel) !== -1;
+    var m;
+
+    GUI.timeout_remove(timer);
+    GUI.timeout_add(timer, function() {
+        builder.state = 'fail';
+        writeToOutput('Failed!<br># ');
+    }, 1000);
+
+    if (isSentinel || builder.state == 'init') {
+        if (builder.state == 'init') {
+            writeToOutput('Building AutoComplete Cache ... ');
+        }
+
+        var stateObject = builder.stateSequence.shift();
+        if (stateObject) {
+            // next state
+            builder.state = stateObject.state;
+            TABS.cli.sendLine(stateObject.command);
+            TABS.cli.sendLine(sentinel);
+        }
+        else {
+            // done
+            cache.settings.sort();
+            cache.commands = Object.keys(cache.commands).sort();
+            cache.resources = Object.keys(cache.resourcesCount).sort();
+            writeToOutput('Done!<br># Press Tab for command suggestions<br># ');
```

Comment:
The texts of the CLI that come from the firmware are in English, so I think maybe is better to let this in English too but I'm not sure. @mikeller what do you think? Is a good idea to translate this?